Modules/EMCAL/src/DigitsQcTask.cxx

```diff
@@ -369,35 +493,25 @@ void DigitsQcTask::DigitsHistograms::clean()
   //  delete h;
   // }
 
-  if (mDigitTime)
-    delete mDigitTime;
-
-  if (mDigitTimeCalib)
-    delete mDigitTimeCalib;
-
-  if (mDigitAmplitude)
-    delete mDigitAmplitude;
-
-  if (mDigitAmplitudeCalib)
-    delete mDigitAmplitudeCalib;
-
-  if (mDigitAmplitudeEMCAL)
-    delete mDigitAmplitudeEMCAL;
-
-  if (mDigitAmplitudeDCAL)
-    delete mDigitAmplitudeDCAL;
-
-  if (mDigitOccupancy)
-    delete mDigitOccupancy;
-
-  if (mDigitOccupancyThr)
-    delete mDigitOccupancyThr;
-
-  if (mIntegratedOccupancy)
-    delete mIntegratedOccupancy;
-
-  if (mnumberEvents)
-    delete mnumberEvents;
+  auto cleanOptional = [](TObject* hist) {
+    if (hist)
+      delete hist;
+  };
+
+  cleanOptional(mDigitTime);
```

Comment:
FYI, deleting nullptr has no effect:
https://stackoverflow.com/questions/6731331/is-it-still-safe-to-delete-nullptr-in-c0x